src/core/linalg/src/dense/4C_linalg_fixedsizematrix_tensor_products.hpp

```diff
@@ -216,6 +216,26 @@ namespace Core::LinAlg::FourTensorOperations
   void add_symmetric_holzapfel_product(Core::LinAlg::Matrix<6, 6>& X,
       const Core::LinAlg::Matrix<3, 3>& A, const Core::LinAlg::Matrix<3, 3>& B, const double fac);
 
+  /*!
+   * @brief Calculates symmetric Holzapfel product
+   *
+   * In tensor index notation this does:
+   * \f[
+   *    X_{abcd} += \text{fac} \cdot \left( A_{ca} \cdot B_{db} + A_{da} \cdot B_{cb} + A_{db}
```

Comment:
You are not adding the contributions, but returning the result. If it is to be added, it has to be done by the user outside.
```suggestion
   *    X_{abcd} = \text{fac} \cdot \left( A_{ca} \cdot B_{db} + A_{da} \cdot B_{cb} + A_{db}
```